ipatests/test_integration/test_sudo.py

```diff
@@ -756,3 +757,1299 @@ def test_domain_resolution_order(self):
+        master.run_command([
+            "ipa", "sudorule-add-runasgroup", self.SUDO_RULE,
+            f"--groups={self.GROUP}"
+        ])
+        clear_sssd_cache(self.client)
+        clear_sssd_cache(master)
+        result = self.run_as_sudo_group(
+            "date", sudo_group=self.GROUP, su_user=self.USER_1)
+
+        result = self.list_sudo_commands(
+            self.USER_1, skip_sssd_cache_clear=True)
+        assert (
+            f"({self.USER_1} : {self.GROUP}) /bin/date"
+            in result.stdout_text
+        )
+
+        stop_ipa_server(master)
+        result = self.run_as_sudo_group(
+            "date", sudo_group=self.GROUP, su_user=self.USER_1,
+            skip_kinit=True, skip_sssd_cache_clear=True)
+
+        self.client.run_command(
+            'su -c "sudo -g %s date" %s' % (self.GROUP, self.USER_1),
+            raiseonerr=False)
+
+        result = self.list_sudo_commands(
+            self.USER_1, skip_kinit=True, skip_sssd_cache_clear=True
+        )
+        assert (
+            f"({self.USER_1} : {self.GROUP}) /bin/date"
+            in result.stdout_text
+        )
+
+        start_ipa_server(master)
+        master.run_command([
+            "ipa", "sudorule-remove-allow-command",
+            "--sudocmds=/bin/date", self.SUDO_RULE
+        ])
+        master.run_command([
+            "ipa", "sudorule-remove-runasgroup", self.SUDO_RULE,
+            f"--groups={self.GROUP}"
+        ])
+        master.run_command([
+            "ipa", "group-remove-member", self.GROUP,
+            f"--users={self.USER_2}"
+        ])
+        master.run_command([
+            "ipa", "group-del", self.GROUP])
+
+    def test_005_sudorule_offline_caching_hostgroup_command(self):
+        master = self.master
+        kinit_admin(master)
+
+        master.run_command([
+            "ipa", "sudorule-add-allow-command", "--sudocmds=/bin/date",
+            self.SUDO_RULE
+        ])
+        master.run_command([
+            "ipa", "hostgroup-add", "--desc=testhostgroup",
+            self.HOST_GROUP
+        ])
+        master.run_command([
+            "ipa", "sudorule-remove-host", self.SUDO_RULE,
+            "--hosts=" + self.client.hostname
+        ])
+        master.run_command([
+            "ipa", "hostgroup-add-member", self.HOST_GROUP,
+            "--hosts=" + self.client.hostname
+        ])
+        master.run_command([
+            "ipa", "sudorule-add-host", self.SUDO_RULE,
+            f"--hostgroup={self.HOST_GROUP}"
+        ])
+        master.run_command([
+            "ipa", "sudorule-add-user", self.SUDO_RULE,
+            "--users", self.USER_1], raiseonerr=False
+        )
+        clear_sssd_cache(self.client)
+        result = self.run_as_sudo_user(
+            "date", sudo_user="root", su_user=self.USER_1)
+        result = self.list_sudo_commands(self.USER_1)
+        assert "(root) /bin/date" in result.stdout_text
+
+        stop_ipa_server(master)
+        result = self.run_as_sudo_user(
```

Comment:
The test doesn't check the result, either use the value in an assertion or replace with self.run_as_sudo_user(